### PR TITLE
add a 'labels-update' command

### PR DESCRIPTION
**Pull request by devoncarew**
Add a 'labels-update' command:
- read the (proposed canonical) labels from dart-lang/.github and print the diffs to a given repo (`--dry-run`)
- or, do the same, but apply the proposed changes (renames, descriptions updates, and label additions) - `--apply-changes`

```
Audit and update the labels used by dart-lang repos.

Usage: report labels-update [arguments] <repo-org/repo-name>
-h, --help             Print this usage information.
    --dry-run          Audit the labels used but don't make any changes to the given repos.
    --apply-changes    Rename, edit, and add labels to bring them in line with those at dart-lang/.github.
                       WARNING: this will make changes to a repo's labels; please preview the changes first by running without '--apply-changes'.

Run "report help" to see global options.
```

### CODE REVIEW

File: pkgs/repo_query/lib/labels_update.dart [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+  return null;
+}
+
+String? wellFormed(String label) {
+  if (allowList.contains(label)) return null;
+
+  if (label.startsWith('package:')) {
+    const avoidPrefix = 'package: ';
+    if (label.startsWith(avoidPrefix)) {
+      return 'rename to package:${label.substring(avoidPrefix.length)}';
```
Comment (mosuem) [inline pkgs/repo_query/lib/labels_update.dart:270]: Is this to filter out the whitespace after the colon? I would then rather write
```
const packagePrefix = 'package:';
if (label.startsWith(packagePrefix)) {
    var labelEnding = label.substring(avoidPrefix.length);
    if (labelEnding.startsWith(r' \s*')) {
      return 'rename to "package:${labelEnding.trim()}"';
```
But it also seems like whitespaces are caught further down, so this case could also simply be removed?

File: pkgs/repo_query/lib/src/common.dart [hunk 2]
```diff
@@ -43,6 +44,10 @@ String iso8601String(DateTime date) {
   return date.toIso8601String().substring(0, 10);
 }
 
+String overflow(String str, [int overflows = 40]) {
+  return str.length <= overflows ? str : '${str.substring(0, overflows)}...';
```
Comment (mosuem) [inline pkgs/repo_query/lib/src/common.dart:48]:
```suggestion
  return str.length <= overflows ? str : '${str.substring(0, overflows - 3)}...';
```

To account for the three periods? :)

**Review by mosuem — APPROVED** [review]

File: pkgs/repo_query/lib/labels_update.dart [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    }
+
+    return 0;
+  }
+}
+
+String? checkSynonym(String label) {
+  for (var entry in synonyms.entries) {
+    if (entry.value.contains(label.toLowerCase())) {
+      return entry.key;
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply pkgs/repo_query/lib/labels_update.dart]: done!

File: pkgs/repo_query/lib/labels_update.dart (hunk 1 above)
Comment (devoncarew) [reply pkgs/repo_query/lib/labels_update.dart]: +1, I'll change to using trim() here for the package name

> But it also seems like whitespaces are caught further down, so this case could also simply be removed?

Here, we're trying to find a good value to rename the label to; below, we're just complaining that they're using a space in the label name.

**Pull request merged**